OneCommunityGlobal/HGNRest
# Haoji no hgn member email subscriptions

# Description
<img width="668" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/99279909/92f9817b-2c91-4cf7-a46a-ac9789ad7fac">

## Related PRS (if any):
To test this backend PR you need to checkout the https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1962 Front end
…

## Main changes explained:
- Create an email sender to broadcast the weekly update to every active user that subscribes email notification
- Non HGN user could subscribe and unsubscribe from our notification
…

## How to test:
Before testing, you need to set up your own test email id to send the mail notification.
This is only for test purposes. Once tested you can revert these changes..

To send an email using the Gmail API with OAuth2 and nodemailer, follow the video below and set it up.
https://www.youtube.com/watch?v=-rcRf7yswfM&t=10s&ab_channel=MafiaCodes

Fill in the these variable values from the above setup in the .env file of the backend.
REACT_APP_EMAIL =
REACT_APP_EMAIL_CLIENT_ID =
REACT_APP_EMAIL_CLIENT_SECRET =
REACT_APP_EMAIL_REFRESH_TOKEN =
REACT_APP_EMAIL_CLIENT_REDIRECT_URI = https://developers.google.com/oauthplayground
sendEmail = true
FRONTEND_URL=http://localhost:3000

FRONTEND_URL is kept updated with current frontend url, in test mode it should be http://localhost:3000

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/99279909/78b32dd1-d5dc-4883-ace9-10e71f806e93


## Note:
see frontend
